test/files/neg/t2462c.check

```diff
@@ -1,7 +1,16 @@
-t2462c.scala:18: error: No C of X$Y
+t2462c.scala:24: error: No C of X$Y
   f[X$Y]
    ^
-t2462c.scala:24: error: No C of Foo[Int]
+t2462c.scala:30: error: No C of Foo[Int]
   f[Foo[Int]]
    ^
-two errors found
+t2462c.scala:33: error: No C of Foo[Int]
+  g[Foo[Int]]
+   ^
+t2462c.scala:36: error: I see no C[Foo[Int]]
+  h[Foo[Int]]
+   ^
+t2462c.scala:40: error: String List[?] List[C[_]] Int Option[Long] -- .
```

Comment:
The `List[?]` is not great here. [The `formatParameterMessage` method](https://github.com/scala/scala/pull/6340/files#diff-7c03397456d3b987549fd039d6b639c6R1657) could be improved for the case of type constructors. cc @adriaanm.

Reply:
consider me sniped :-) I pushed a commit that adds some ad-hoc pretty printing for these. Feel free to `g push -f HEAD^